kernel/src/hil/digest.rs

```diff
@@ -56,13 +108,70 @@ pub trait Digest<'a, const L: usize> {
     /// error with error code ENOSUPPORT.
     fn run(&'a self, digest: &'static mut [u8; L])
         -> Result<(), (ErrorCode, &'static mut [u8; L])>;
+}
 
-    /// Clear the keys and any other sensitive data.
-    /// This won't clear the buffers provided to this API, that is up to the
-    /// user to clear.
-    fn clear_data(&self);
+/// Computes a digest (cryptographic hash) over data
+///
+/// 'L' is the length of the 'u8' array to store the digest output.
+pub trait DigestVerify<'a, const L: usize> {
+    /// Set the client instance which will receive the `verification_done()`
+    /// callback.
+    /// This is not required is using the `set_client()` fuction from the
+    /// `Digest` trait.
+    #[allow(unused_variables)]
+    fn set_verify_client(&'a self, client: &'a dyn ClientVerify<'a, L>) {}
+
+    /// Compare the specified digest in the `compare` buffer to the calculated
+    /// digest. This function is similar to `run()` and should be used instead
+    /// of `run()` if the caller doesn't need to know the output, just if it
+    /// matches a known value.
+    ///
+    /// For example:
+    /// ```rust,ignore
+    ///     // Compute a digest on data
+    ///     add_data(...);
+    ///     add_data(...);
+    ///
+    ///     // Compare the computed digest generated to an existing digest
+    ///     verify(...);
+    /// ```
+    /// NOTE: The above is just pseudo code. The user is expected to check for
```

Comment:
If it's just pseudocode, perhaps make it not rust formatted?

Reply:
Yeah, I'll remove that then. I like the syntax highlighting though :)